src/System.CommandLine/Argument{T}.cs

```diff
@@ -18,9 +18,9 @@ public Argument() : base(null)
             ArgumentType = typeof(T);
         }
 
-        public Argument(string name, T defaultValue) : this(name)
+        public Argument(string name, Func<T> defaultValue) : this(name)
         {
-            SetDefaultValue(defaultValue);
+            SetDefaultValue(() => defaultValue());
```

Comment:
Is there a reason we are wrapping the Func in another Func?
```suggestion
            SetDefaultValue(defaultValue);
```

Reply:
`SetDefaultvalue` has an overload that takes `object`. 

A cast could also work here.

Reply:
ah I missed that. I would suggest the cast here (and in the few other usages in this file) to avoid extra allocations.